crates/rattler_package_streaming/src/archive.rs

```diff
@@ -0,0 +1,194 @@
+//! This crate provides the ability to extract a specified directory from tar.bz2 or conda archive.
+
+use std::path::{Path, PathBuf};
+
+use rattler_conda_types::package::ArchiveType;
+
+use crate::fs::{extract_directory_from_conda, extract_directory_from_tar_bz2};
+
+/// A struct representing an archive file located on disk.
+pub struct LocalArchive {
+    /// Archive type representing the type of archive.
+    pub archive_type: ArchiveType,
+    /// Location of the archive file on disk.
+    pub location: PathBuf,
+}
+
+impl LocalArchive {
+    /// Extracts the contents of the archive to the specified destination.
+    pub fn extract_a_folder(
```

Comment:
Can we also rename this to match the function it calls? 

```suggestion
    pub fn extract_directory(
```